mvc/views/dependency-injection.rst

```diff
@@ -1,93 +1,79 @@
+
+.. literalinclude:: dependency-injection/sample/src/ViewInjectSample/Views/Todo/Index.cshtml
   :linenos:
+  :language: c#
+  :emphasize-lines: 4-5,15-17
+
+This view displays a list of ``ToDoItem`` instances, along with a summary showing overall statistics. The summary is populated from the injected ``StatisticsService``. This service is registered for dependency injection in ``ConfigureServices`` in ``Startup.cs``:
+
+.. literalinclude:: dependency-injection/sample/src/ViewInjectSample/Startup.cs
+  :linenos:
+  :lines: 15-22
+  :language: c#
+  :emphasize-lines: 5-6
+  :dedent: 8
   
-  @inject TodoList.Services.StatisticsService Statistics
-
-3. Add markup calling the StatisticsService to the end of the file:
-
-  .. code-block:: html
-    :linenos:
-    :emphasize-lines: 6-11
-    
-    @* Markup removed for brevity *@
-    <div>@Html.ActionLink("Create New Todo", "Create", "Todo") </div>
-    </div>
-      <div class="col-md-4">
-        @await Component.InvokeAsync("PriorityList", 4, true)
-        <h3>Stats</h3>
-        <ul>
-          <li>Items: @await Statistics.GetCount()</li>
-          <li>Completed:@await Statistics.GetCompletedCount()</li>
-          <li>Average Priority:@await Statistics.GetAveragePriority()</li>
-        </ul>
-      </div>
-    </div>
-
-4. Register the ``StatisticsService`` class in the *Startup.cs* file: 
-
-  .. code-block:: c#
-    :linenos:
-    :emphasize-lines: 8
-    
-    public void ConfigureServices(IServiceCollection services)
-    {
-      // Code removed for brevity.
-      // Add MVC services to the services container.
-      services.AddMvc();
-      services.AddTransient<TodoList.Services.StatisticsService>();
-    }
-
-The statistics are displayed:
- 
-.. image:: dependency-injection/_static/stat.png
+The ``StatisticsService`` simply performs some calculations on the set of ``ToDoItem`` instances, which it accesses via a repository:
+
+
+.. literalinclude:: dependency-injection/sample/src/ViewInjectSample/Model/Services/StatisticsService.cs
+  :linenos:
+  :language: c#
+  :emphasize-lines: 16,21,27
+
+.. note:: In this sample the repository uses an in-memory collection. The implementation shown above is not recommended for large, remotely accessed data sets.
+
+The end result of this approach is the display of data from both the model bound to the view as well as the service injected into the view:
+
+.. image:: dependency-injection/_static/screenshot.png
+
+Populating Lookup Data
+----------------------
+One example where you may want to use view injection is population of options for certain UI elements, such as dropdown lists. Consider a user profile form that includes options for specifying gender, state, and other preferences. Rendering such a form using a standard MVC approach would require the controller to request data access services for each of these sets of options, and then populate a model or ``ViewBag`` with each set of options to be bound.
```

Comment:
View injection can be useful to populate options in UI elements, such as dropdown lists.